src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.RegularCompilationTracker_Generators.cs

```diff
@@ -94,15 +116,15 @@ private sealed partial class RegularCompilationTracker : ICompilationTracker
             using var _ = await RemoteKeepAliveSession.CreateAsync(compilationState, cancellationToken).ConfigureAwait(false);
 
             // First, grab the info from our external host about the generated documents it has for this project.  Note:
-            // we ourselves are the innermost "RegularCompilationTracker" responsible for actually running generators.
-            // As such, our call to the oop side reflects that by asking for the real source generated docs, and *not*
-            // any overlaid 'frozen' source generated documents.
+            // we ourselves are the innermost "RegularCompilationTracker" responsible for actually running generators. If
+            // we're in a createOnlyRequired state, we'll only run the required generators, and not get results for any other
+            // generators.
             var projectId = this.ProjectState.Id;
             var infosOpt = await connection.TryInvokeAsync(
                 compilationState,
                 projectId,
                 (service, solutionChecksum, cancellationToken) => service.GetSourceGeneratedDocumentInfoAsync(
-                    solutionChecksum, projectId, withFrozenSourceGeneratedDocuments: false, cancellationToken),
+                    solutionChecksum, projectId, withFrozenSourceGeneratedDocuments: false, forceOnlyRequiredDocuments: creationPolicy == GeneratedDocumentCreationPolicy.CreateOnlyRequired, cancellationToken),
```

Comment:
Can we just send the CreationPolicy across the wire rather than decomposing to a boolean? Otherwise seems like this is easy to miss if we were to add another state.